src/parser/shaman/enhancement/modules/core/MaelstromWeapon.tsx

```diff
@@ -0,0 +1,231 @@
+      this.MAX_STACKS_SPENT_PER_CAST
+    );
+
+    if (stacksUsed === 0) {
+      return;
+    }
+
+    this.stacksUsed += stacksUsed;
+
+    if (this.currentStacks === this.MAX_STACKS) {
+      this.stopHavingMaxStacks(event.timestamp);
+    }
+
+    this.currentStacks -= stacksUsed;
+  }
+
+  gainFirstStack(event: ApplyBuffEvent) {
+    this.currentStacks = 1;
+    this.stacksGained += 1;
+  }
+
+  gainSubsequentStack(event: ApplyBuffStackEvent) {
+    if (this.currentStacks + 1 !== event.stack) {
+      console.error('Maelstrom Weapon analyzer: stack mismatch in applybuffstack');
+      return;
+    }
+
+    this.currentStacks = event.stack;
+    this.stacksGained += 1;
+
+    if (this.currentStacks === this.MAX_STACKS) {
+      this.reachMaxStacks(event.timestamp)
+    }
+  }
+
+  removeAllStacks(event: RemoveBuffEvent) {
+    if (this.currentStacks === 0) {
+      // stacks were spend on a spender spell
+      return;
+    }
+
+    if (this.currentStacks === 10) {
+      this.stopHavingMaxStacks(event.timestamp);
+    }
+
+    this.stacksExpired += this.currentStacks;
+    this.currentStacks = 0;
+  }
+
+
+  spendStacks(event: RemoveBuffStackEvent) {
+    if (this.currentStacks !== event.stack) {
+      console.error('Maelstrom Weapon analyzer: stack mismatch in spendStacks')
+    }
+    if (this.currentStacks === this.MAX_STACKS) {
+      this.stopHavingMaxStacks(event.timestamp);
+    }
+
+    this.currentStacks = event.stack;
+  }
+
+  statistic() {
+    return <Statistic
+      position={STATISTIC_ORDER.CORE()}
+      size="flexible"
+      tooltip={<>
+        You gained {this.stacksGained} Maelstrom Weapon stacks and used {this.stacksUsed}
+      </>}
+    >
+      <BoringSpellValueText spell={SPELLS.MAELSTROM_WEAPON}>
+        <UptimeIcon /> {formatPercentage(this.timePercentageSpentWithCappedStacks)}% <small>of fight with max stacks</small><br />
+        {formatPercentage(this.stacksUsed / this.stacksGained)}% <small>of stacks used</small>
+      </BoringSpellValueText>
+    </Statistic>
+  }
+}
+
+/**
+ * This is used to calculate the amount of time spent while having max number of Maelstrom Weapon stacks
+ */
+class Interval {
+  protected startTime: number;
+  protected endTime: number | undefined = undefined;
+
+  constructor(timestamp: number) {
+    this.startTime = timestamp;
+  }
+
+  end(timestamp: number) {
+    this.endTime = timestamp;
+  }
+
+  get duration(): number {
+    if (this.endTime === undefined) {
+      console.error('Cannot calculate duration of an Interval with no endTime.');
+      return 0;
+    }
+    return this.endTime - this.startTime;
+  }
+
+  get ended(): boolean {
+    return this.endTime !== undefined;
+  }
+}
+
+class Intervals {
+  protected intervals: Interval[];
+
+  constructor() {
+    this.intervals = [];
+  };
+
+  startInterval(timestamp: number) {
+    if (this.isLastIntervalInProgress) {
+      console.log('Intervals: cannot start a new interval because one is already in progress.')
```

Comment:
console logs should only be run if a "debug mode" is enabled.

So you can do 'debug && console.log(...' and add a debug = false constant, or remove the logs.